.github/workflows/python-docs.yml

```diff
@@ -0,0 +1,27 @@
+name: documentation (python)
+
+on: [push, pull_request, workflow_dispatch]
+
+permissions:
+  contents: write
+
+jobs:
+  docs:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+      - name: Install dependencies
+        run: |
+          pip install sphinx sphinx_rtd_theme myst_parser
```

Comment:
we should add these dev dependencies to the root pyproject.toml so they are versioned, and install with poetry (see the main python.yml workflow)